### DISCUSSION TITLE
Workflow of Apache NIFI with respect to reading data to Database

### DISCUSSION THREAD

**Discussion by GitVivekHub** [Sunbird-cQube/community#814] [Q&A]
Me and my team are trying to ingest custom data into cqube with our own dimension and event files. We are able to create the schema and upload data to the minio bucket. However when we want to read and insert the CSV files uploaded into the database we are unable to understand what changes are to be done to the python script and where exactly it is to be done.

It would be better if anyone from your team can assist with python scripts that are responsible for creating dataset and dimensions schema along with its tables

**Comment by RNarmatha** [comment]
Rajesh please look into this issue

**Reply by bommalla** [reply]
@GitVivekHub Please follow the below steps to configure the Apache nifi
step1: Please open below mentioned python path
https://github.com/Sunbird-cQube/generator-ms/blob/dev/static_processor_group/add_nifi_template.py

step2: In the python file line number 595 change the dimension path

step3: In the python file line number 613 change the programs path

step4: After that create one config.ini file and put below mentioned details over there

```[CREDs]

storage_type = local
nifi_host = http://localhost
nifi_port = 8080
state_name = JH
instance_type = VSK
processing_time = * * * * * ?
plugin_time = * * * * * ?
adapter_schedule_time = * * * * * ?
minio_end_point = 
minio_port = 
minio_bucket = 
minio_access_key =
minio_secret_key =

step5: Then navigate the below mentioned path
https://github.com/Sunbird-cQube/generator-ms/tree/dev/static_processor_group

step6: Then copy the config.ini file into above mentioned path
Ex: https://github.com/Sunbird-cQube/generator-ms/tree/dev/static_processor_group/config.ini

step7: Then run the below mentioned python file 
python3 add_nifi_template.py

**Reply by GitVivekHub** [reply]
Hello @bommalla @RNarmatha i tried the above steps, here my apache nifi is configured now for next step to add  my custom programs that is my custom dimension and event files what should i consider?

**Reply by bommalla** [reply]
@GitVivekHub Please find the below document and move forward to configure the new programs.

https://cqube.sunbird.org/ui-configurations/adding-a-new-program
